Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -70,97 +88,648 @@ class UpstreamIntegrationTests: XCTestCase {
+    }
+
+    /// Demonstrates flexible validation using general wildcard match but only on exact match key paths; shows example failure message
+    ///
+    /// See `testLocationHint_onlyExpectedKeys_usingGeneralWildcard` for use cases covered
+    func testLocationHint_onlyExpectedKeys_usingGeneralWildcard_missingMatchShouldFail() {
+        let expectedJSON = #"""
+        {
+          "payload": [
+            {
+              "scope" : "EdgeNetwork"
+            },
+            {
+              "scope" : "Target"
+            }
+          ]
+        }
+        """#
+
+        let actualJSON = #"""
+        {
+         "payload": [
+           {
+             "ttlSeconds" : 1800,
+             "scope" : "Target",
+             "hint" : "35"
+           },
+           {
+             "ttlSeconds" : 1800,
+             "scope" : "AAM",
+             "hint" : "9"
+           },
+           {
+             "ttlSeconds" : 1800,
+             "scope" : "EdgeNetwork_DIFFERENT",
+             "hint" : "or2"
+           }
+         ]
+        }
+        """#
+
+        guard let expected = getAnyCodable(expectedJSON), let actual = getAnyCodable(actualJSON) else {
+            XCTFail("Unable to decode JSON string. Test case unable to proceed.")
+            return
+        }
+
+        assertContains(defaultMode: .typeMatch, expected: expected, actual: actual, alternateModePaths: ["payload[*].scope"])
+    }
+
+    /// Demonstrates flexible validation using general wildcard match but only on exact match key paths
+    /// Use cases covered:
+    /// 1. Array general wildcard matching: exact match path -> `[*]`
+    /// 2. Array subset matching: only 2/3 on the expected side
+    /// 3. Exact match validation on only specific subset of keys in a dictionary: exact match path -> `.scope`
+    func testConsentUpdate_onlyExpectedKeys_usingGeneralWildcard() {
+        let expectedJSON = #"""
+        {
+          "meta" : {
+            "konductorConfig" : {
+              "streaming" : {
+                "enabled" : true,
+                "recordSeparator" : "\u0000",
+                "lineFeed" : "\n"
+              }
+            }
+          },
+          "query" : {
+            "consent" : {
+              "operation" : "update"
+            }
+          },
+          "identityMap" : {
+            "ECID" : [
+              {
+                "id" : "01568806327147089148132339481432735151",
+                "authenticatedState" : "ambiguous",
+                "primary" : false
+              }
+            ]
+          },
+          "consent" : [
+            {
+              "standard" : "Adobe",
+              "version" : "2.0",
+              "value" : {
+                "collect" : {
+                  "val" : "n"
+                },
+                "metadata" : {
+                  "time" : "2023-04-21T21:56:52.455Z"
+                }
+              }
+            }
+          ]
+        }
+        """#
+
+        let actualJSON = #"""
```

Comment:
I assume we will take the actualJSON from the network request body for these tests?